movement/mover.py

```diff
@@ -197,18 +198,19 @@ def volume_ratio(self):
         :return: the ratio of the largest and smallest element volumes.
         :rtype: :class:`float`
         """
-        volume_array = self.volume.vector().gather()
-        return volume_array.max() / volume_array.min()
+        return function_data_max(self.volume) / function_data_min(self.volume)
 
     @property
     def coefficient_of_variation(self):
         """
         :return: the coefficient of variation (σ/μ) of element volumes.
         :rtype: :class:`float`
         """
-        volume_array = self.volume.vector().gather()
-        mean = volume_array.sum() / volume_array.size
-        return np.sqrt(np.sum((volume_array - mean) ** 2) / volume_array.size) / mean
+        size = self.volume.dat.shape[0]
```

Comment:
Oh hm, I think this changes things. Before we used `volume_array.size` which is the size of the gathered vector, whereas `.dat.shape[0]` is just the shape of the local (including halo) domain. It seems internally firedrake typically uses `.dat.dataset.layout_vec.getSizes()` to get the global size. Maybe we should add a `function_data_avg()` to animate?

Reply:
Sorry, could you explain why we need `function_data_avg()`?

Reply:
Ah ignore that, I just thought you could wrap this, calculating the mean of the global data vector, in a helper function to avoid having this low level stuff in many places - but it seems to be the only place